src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java

```diff
@@ -145,19 +147,33 @@ public BucketMetadata getMetadata(
             @ApiParam(value = "The name of objects that buckets must contain") @RequestParam(value = "objectName", required = false) Optional<String> objectName)
             throws NotAuthenticatedException, AccessDeniedException {
 
+        List<BucketMetadata> listBucket;
+        log.debug("====== Get buckets list request started ======== ");
+        long startTime = System.currentTimeMillis();
+
+        //transform empty String into an empty Optional
+        kind = kind.filter(s -> !s.isEmpty());
+        contentType = contentType.filter(s -> !s.isEmpty());
+        objectName = objectName.filter(s -> !s.isEmpty());
         if (sessionIdRequired) {
             RestApiAccessResponse restApiAccessResponse = restApiAccessService.checkAccessBySessionIdForOwnerOrGroupAndThrowIfDeclined(sessionId,
                                                                                                                                        ownerName);
-
-            return bucketService.getBucketsByGroups(ownerName,
-                                                    kind,
-                                                    contentType,
-                                                    objectName,
-                                                    () -> restApiAccessResponse.getAuthenticatedUser().getGroups());
+            log.debug("Check Access By Session Id For Owner Or Group " + (System.currentTimeMillis() - startTime) +
+                      " ms.");
+            listBucket = bucketService.getBucketsByGroups(ownerName,
+                                                          kind,
+                                                          contentType,
+                                                          objectName,
+                                                          () -> restApiAccessResponse.getAuthenticatedUser()
+                                                                                     .getGroups());
 
         } else {
-            return bucketService.listBuckets(ownerName, kind, contentType, objectName);
+            listBucket = bucketService.listBuckets(ownerName, kind, contentType, objectName);
         }
+        log.debug("The whole controller request to Get bucket lists done in " +
```

Comment:
regarding debug log, it would be more readable to write:
log.debug("bucket list timer : validate session : " + time + " ms")
log.debug("bucket list timer : total : " + time + " ms")

basically identify the request, then each timer